lib/spark.js

```diff
@@ -202,6 +217,29 @@ export default class Spark {
     }
   }
 
+  async testNetworkRetrieval(providers, cid, stats) {
+    if (!providers.length) {
+      console.info('No alternative providers found for the CID.')
+      return
+    }
+
+    stats.networkRetrieval = newNetworkRetrievalStats()
+    const randomProvider = pickRandomProvider(providers)
```

Comment:
I'm surprised by this strategy. I might be out of sync with the current product plan, but wasn't it the goal to see if _any_ provider can fetch the data? Here we're marking a failure if the 2nd provider we try fails, but don't try the other ones.

Reply:
I've posted my opinions on that in this [comment](https://github.com/CheckerNetwork/roadmap/issues/254#issuecomment-2786128738) and I'm happy to discuss more! 

Other motivation behind this was not to put too much pressure on the network serving content and on the checkers. I'm afraid of the worst case situation where we'd have to check every provider serving content.